index.js

```diff
@@ -95,11 +96,14 @@ Request.prototype._tryUntilFail = function () {
   this.attempts++;
 
   this._req = Request.request(this.options, function (err, response, body) {
+    var delayStrategyResult = this.delayStrategy ? this.delayStrategy(err, response, body) : undefined;
+    var delayStrategyResultValid = typeof delayStrategyResult === "number";
+
     if (response) {
       response.attempts = this.attempts;
     }
     if (this.retryStrategy(err, response, body) && this.maxAttempts > 0) {
-      this._timeout = setTimeout(this._tryUntilFail.bind(this), this.retryDelay);
+      this._timeout = setTimeout(this._tryUntilFail.bind(this), delayStrategyResultValid ? delayStrategyResult : this.retryDelay);
```

Comment:
I'm fairly new to JS. Why did you choose not to type check and perhaps alert the user early? In my type check I probably should have alerted the user as well instead of falling back to the retryDelay.